src/wallet/rpcwallet.cpp

```diff
@@ -3594,17 +3604,20 @@ UniValue z_listunifiedreceivers(const UniValue& params, bool fHelp)
 }
 
 CAmount getBalanceTaddr(const std::optional<CTxDestination>& taddr, const std::optional<int>& asOfHeight, int minDepth=1, bool ignoreUnspendable=true) {
-    vector<COutput> vecOutputs;
     CAmount balance = 0;
 
     LOCK2(cs_main, pwalletMain->cs_wallet);
 
-    pwalletMain->AvailableCoins(vecOutputs, asOfHeight, false, NULL, true);
-    for (const COutput& out : vecOutputs) {
-        if (out.nDepth < minDepth) {
-            continue;
-        }
+    vector<COutput> vecOutputs = pwalletMain->AvailableCoins(
+            asOfHeight,
+            false,        // fOnlyConfirmed
+            nullptr,      // coinControl
+            true,         // fIncludeZeroValue
+            true,         // fIncludeCoinBase
+            ignoreUnspendable ? ISMINE_LEGACY_SPENDABLE : ISMINE_LEGACY_ALL,
```

Comment:
```suggestion
            ISMINE_LEGACY_SPENDABLE,
            false,        // fOnlySpendable
```
although in this case the unspendable outputs are all skipped, so this could also be:
```suggestion
            ISMINE_LEGACY_SPENDABLE,
            ignoreUnspendable,
```
and then the skipping code below could be removed.